Detectors/AOD/src/AODProducerWorkflowSpec.cxx

```diff
@@ -194,12 +196,86 @@ void AODProducerWorkflowDPL::addToMFTTracksTable(mftTracksCursorType& mftTracksC
+  const auto& tpcClusShMap = data.clusterShMapTPC;
+  const auto& tpcClusAcc = data.getTPCClusters();
+
+  for (int src = GIndex::NSources; src--;) {
+    int start = trackRef.getFirstEntryOfSource(src);
+    int end = start + trackRef.getEntriesOfSource(src);
+    LOG(DEBUG) << "Unassigned tracks: src = " << src << ", start = " << start << ", end = " << end;
+    for (int ti = start; ti < end; ti++) {
+      TrackExtraInfo extraInfoHolder;
+      auto& trackIndex = GIndices[ti];
+      if (GIndex::includesSource(src, mInputSources)) {
+        if (src == GIndex::Source::MFT) { // MFT tracks are treated separately since they are stored in a different table
+          const auto& track = data.getMFTTrack(trackIndex.getIndex());
+          addToMFTTracksTable(mftTracksCursor, track, collisionID);
+        } else {
+          auto contributorsGID = data.getSingleDetectorRefs(trackIndex);
+          const auto& trackPar = data.getTrackParam(trackIndex);
+          if (contributorsGID[GIndex::Source::ITS].isIndexSet()) {
+            const auto& itsOrig = data.getITSTrack(contributorsGID[GIndex::ITS]);
+            extraInfoHolder.itsClusterMap = itsOrig.getPattern();
+          }
+          if (contributorsGID[GIndex::Source::TPC].isIndexSet()) {
+            const auto& tpcOrig = data.getTPCTrack(contributorsGID[GIndex::TPC]);
+            extraInfoHolder.tpcInnerParam = tpcOrig.getP();
+            extraInfoHolder.tpcChi2NCl = tpcOrig.getNClusters() ? tpcOrig.getChi2() / tpcOrig.getNClusters() : 0;
+            extraInfoHolder.tpcSignal = tpcOrig.getdEdx().dEdxTotTPC;
+            uint8_t shared, found, crossed; // fixme: need to switch from these placeholders to something more reasonable
+            countTPCClusters(tpcOrig, tpcClusRefs, tpcClusShMap, tpcClusAcc, shared, found, crossed);
+            extraInfoHolder.tpcNClsFindable = tpcOrig.getNClusters();
+            extraInfoHolder.tpcNClsFindableMinusCrossedRows = tpcOrig.getNClusters() - crossed;
+            extraInfoHolder.tpcNClsShared = shared;
+          }
+          if (contributorsGID[GIndex::Source::ITSTPCTOF].isIndexSet()) {
+            const auto& tofMatch = data.getTOFMatch(contributorsGID[GIndex::Source::ITSTPCTOF]);
+            extraInfoHolder.tofChi2 = tofMatch.getChi2();
+            const auto& tofInt = tofMatch.getLTIntegralOut();
+            float intLen = tofInt.getL();
+            extraInfoHolder.length = intLen;
+            float mom = trackPar.getP();
+            float mass = o2::constants::physics::MassPionCharged; // default pid = pion
+            float expSig = mom > 0 ? std::sqrt(mom * mom + mass * mass) : 0.f;
+            extraInfoHolder.tofSignal = expSig;
+            float expMom = 0.f;
+            if (expSig > 0) {
+              float expBeta = (intLen / expSig / cSpeed);
+              expMom = mass * expBeta * cSpeed / std::sqrt(1.f - (expBeta * expBeta));
+            }
+            extraInfoHolder.tofExpMom = expMom;
```

Comment:
This does not look correct:
1) the `extraInfoHolder.tofSignal` should be the the measured time of flight rather than the energy. As in the current `o2::dataformats::MatchInfoTOF` there is no `getSignal()` method, the best if @noferini hints how it should be extracted (I think corresponding data member should be added).
The `extraInfoHolder.tofExpMom` should be calculated as `mass * expBeta / sqrt(1. - expBeta*expBeta) `, the `expBeta` will be correct provided the expSig is fixed. I've asked @njacazio to verify this by it seems he is offline.

Reply:
PS: details of TOF parameters are here: https://indico.cern.ch/event/1010843/contributions/4242241/attachments/2193866/3708654/jacazio_2021-02-22_O2PID.pdf

Reply:
I see. I will be waiting for the response then

Reply:
The TOF time is not propagated in the MatchingInfo but it can be recovever from the cluster (tofclusters.root) with position
tofMatch.getTOFClIndex() 
If you think it is better to propagate this infromation in the matching info I can add the data member and then the proper method. Do you need it?

Reply:
By accepting this
https://github.com/nburmaso/AliceO2/pull/1
you could simply do
            extraInfoHolder.tofSignal = tofMatch.getSignal();

Reply:
Please, note that the interaction time is not subtracted (you get a time inside the timeframe)

Reply:
Hi @shahor02. I have fixed `tofExpMom` calculation, could you please check if it's ok?